src/main/java/ch/njol/skript/lang/Expression.java

```diff
@@ -175,6 +175,31 @@ default Optional<T> getOptionalSingle(Event event) {
 	 */
 	Class<? extends T> getReturnType();
 
+	/**
+	 * For expressions that might return multiple (incalculable at parse time) types,
+	 * this provides a list of all possible types.
+	 * Use cases include: expressions that depend on the return type of their input.
+	 *
+	 * @return A list of all possible types this might return
+	 */
+	default Class<? extends T>[] possibleReturnTypes() {
```

Comment:
im wondering if this needs an implementation on implementations such as ExpressionList (which currently uses a get super class call)

Reply:
I did it for Literal lists too.